cmd/grype/cli/commands/root_test.go

```diff
@@ -86,3 +89,118 @@ func Test_getProviderConfig(t *testing.T) {
 		})
 	}
 }
+
+func Test_applyVexRules(t *testing.T) {
+	tests := []struct {
+		name                   string
+		initialIgnoreRules     []match.IgnoreRule
+		vexDocuments           []string
+		vexAdd                 []string
+		expectedIgnoreRules    []match.IgnoreRule
+		expectError            bool
+		expectedErrorSubstring string
+	}{
+		{
+			name:                "no VEX documents provided - no rules added",
+			initialIgnoreRules:  []match.IgnoreRule{},
+			vexDocuments:        []string{},
+			vexAdd:              []string{},
+			expectedIgnoreRules: []match.IgnoreRule{},
+			expectError:         false,
+		},
+		{
+			name:               "VEX documents provided with empty ignore rules - automatic rules added",
+			initialIgnoreRules: []match.IgnoreRule{},
+			vexDocuments:       []string{"path/to/vex.json"},
+			vexAdd:             []string{},
+			expectedIgnoreRules: []match.IgnoreRule{
+				{VexStatus: string(vex.StatusNotAffected)},
+				{VexStatus: string(vex.StatusFixed)},
+			},
+			expectError: false,
+		},
+		{
+			name: "VEX documents provided with existing ignore rules - automatic rules still added",
+			initialIgnoreRules: []match.IgnoreRule{
+				{Vulnerability: "CVE-2023-1234"},
+			},
+			vexDocuments: []string{"path/to/vex.json"},
+			vexAdd:       []string{},
+			expectedIgnoreRules: []match.IgnoreRule{
+				{Vulnerability: "CVE-2023-1234"},
+				{VexStatus: string(vex.StatusNotAffected)},
+				{VexStatus: string(vex.StatusFixed)},
+			},
+			expectError: false,
+		},
+		{
+			name:               "vex-add with valid statuses",
+			initialIgnoreRules: []match.IgnoreRule{},
+			vexDocuments:       []string{"path/to/vex.json"},
+			vexAdd:             []string{"affected", "under_investigation"},
+			expectedIgnoreRules: []match.IgnoreRule{
+				{VexStatus: string(vex.StatusNotAffected)},
+				{VexStatus: string(vex.StatusFixed)},
+				{VexStatus: string(vex.StatusAffected)},
+				{VexStatus: string(vex.StatusUnderInvestigation)},
+			},
+			expectError: false,
+		},
+		{
+			name:                   "vex-add with invalid status",
+			initialIgnoreRules:     []match.IgnoreRule{},
+			vexDocuments:           []string{"path/to/vex.json"},
+			vexAdd:                 []string{"invalid_status"},
+			expectedIgnoreRules:    nil,
+			expectError:            true,
+			expectedErrorSubstring: "invalid VEX status in vex-add setting: invalid_status",
+		},
+		{
+			name:                   "vex-add attempting to use fixed status",
```

Comment:
This test preserves existing behavior, namely that putting "fixed" in vex-add is an error. Really, it's redundant, so I'm open to changing this to a no-op rather than an error.